jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -654,32 +697,299 @@ def _set_value_in_subset_definition(self, index, name, desired_key, new_value):
                 self.subset_definitions[index][i]['value'] = new_value
                 break
 
-    def _import_spectral_regions(self, spec_region, mode=NewMode, viewer=None):
+    def import_region(self, region, combination_mode=None, max_num_regions=None,
+                      refdata_label=None, return_bad_regions=False, **kwargs):
         """
-        Method for importing a SpectralRegion object or list of SpectralRegion objects.
+        Method for creating subsets from regions or region files.
 
         Parameters
         ----------
-        spec_region : ~`specutils.SpectralRegion` object or list
-            The object that contains bounds for the spectral region or regions
-        mode : AndMode, AndNotMode, OrMode, ReplaceMode, XorMode, NewMode, or list
-            By default this is set to NewMode which creates a new subset per subregion.
-            OrMode will create a composite subset with each subregion corresponding
-            to a subregion of a single subset.
+        region : region, list of region objects, or str
+            A region object can be one of the following:
+
+            * Astropy ``regions`` object
+            * ``photutils`` apertures (limited support until ``photutils``
+              fully supports ``regions``)
+            * specutils ``SpectralRegion`` object
+
+            A string which represents a ``regions`` or ``SpectralRegion`` file.
+            If given as a list, it can only contain spectral or non-spectral regions, not both.
+
+        combination_mode : list, str, or `None`
+            The way that regions are created or combined. If a list, then it must be the
+            same length as regions. If `None`, then it will follow the default glue
+            functionality for subset creation.
```

Comment:
could/should we restore the original combination mode after this either completes or raises an exception?

Reply:
My thought would be the combination mode stays the last entry in the list but I am interested to hear what would make the most sense for a user.

Reply:
ok, let's discuss and decide with the team.  Whatever the decision, its probably worth explaining here in the API docs so the users know what to expect.